### DISCUSSION TITLE
Make custom buttons (rect) and trigger when clicked

### DISCUSSION THREAD

**Discussion by peterpranger** [cables-gl/cables_docs#630] [Answers & Questions] [answered]
Hello everyone!
I'm really new to Cables and struggling with some very basic interactions.
What I trying to achieve:
Loading a 3d scene with camera (views)(this works)
Creating custom buttons, assigning each button to a camera view. So when I click button1 the 3D scene uses camera1, When I click button2 the 3D scene switches to camera 2 etc.

I got some parts working her (it's a mess):
https://cables.gl/p/BXE6MM

I really like the potential of Cables!!!
Cheers,
Peter

**Comment by peterpranger** [comment]
Hey !
I think i've got something:
https://cables.gl/edit/zitKQM

**Comment by pandrr** [comment, marked as answer]
there are many ways to make things interactive.
every one has its pros and cons

- for 2d stuff i would always try to use html elements (pro: independant rendering and handling by the browser and css)

https://dev.cables.gl/op/Ops.Html.DivElement_v3

- intersect ops (performant way for simple 3d volumes)

https://cables.gl/op/Ops.Graphics.Intersection.IntersectWorld
https://www.youtube.com/watch?v=lNbfCp0T55c


- interactive recatangle, you found that already

picking by rendering specific colors into a texture:
https://www.youtube.com/watch?v=WQM37Jr_NHI
